cylc/flow/platforms.py

```diff
@@ -265,6 +266,12 @@ def platform_from_name(
             platform_data['name'] = platform_name
             return platform_data
 
+    # If platform name in run mode and not otherwise defined:
+    if platform_name in RunMode.WORKFLOW_MODES:
+        platform_data = deepcopy(platforms['localhost'])
+        platform_data['name'] = platform_name
+        return platform_data
```

Comment:
Agree with all of this, except the bit about dummy mode. This doesn't affect dummy mode, which does, and always has (at least since dummy-local was rebranded as dummy) used localhost.